packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2315,50 +2316,156 @@ class ConstrainedBox extends SingleChildRenderObjectWidget {
   }
 }
 
-/// A widget that imposes no constraints on its child, allowing it to render
-/// at its "natural" size.
+/// A container widget that applies an arbitrary transform to its constraints,
+/// and sizes its child using the resulting [BoxConstraints], treating any
+/// overflow as error.
 ///
-/// This allows a child to render at the size it would render if it were alone
-/// on an infinite canvas with no constraints. This container will then attempt
-/// to adopt the same size, within the limits of its own constraints. If it ends
-/// up with a different size, it will align the child based on [alignment].
-/// If the box cannot expand enough to accommodate the entire child, the
-/// child will be clipped.
+/// This container sizes its child using a [BoxConstraints] created by applying
+/// [constraintsTransform] to its own constraints. This container will then
+/// attempt to adopt the same size, within the limits of its own constraints. If
+/// it ends up with a different size, it will align the child based on
+/// [alignment]. If the container cannot expand enough to accommodate the entire
+/// child, the child will be clipped if [clipBehavior] is not [Clip.none].
 ///
 /// In debug mode, if the child overflows the container, a warning will be
 /// printed on the console, and black and yellow striped areas will appear where
 /// the overflow occurs.
 ///
+/// When [child] is null, this widget becomes as small as possible and never
+/// overflows
+///
+/// This widget can be used to ensure some of [child]'s intrinsic dimensions are
+/// honored, during development. For instance, if [child] requires a minimum
```

Comment:
What do you mean by "during development"? Should this widget not be used in production?

Reply:
changed to "This widget can be used to ensure some of [child]'s natrual dimensions are honored, and get an early warning otherwise during development. "